backend/inspirehep/editor/cache.py

```diff
@@ -0,0 +1,89 @@
+from datetime import datetime
+
+import structlog
+from flask import current_app
+from redis import StrictRedis
+
+from inspirehep.config import REDIS_KEY_VALUE_PAIR_EXPIRE_TIME
+
+LOGGER = structlog.getLogger()
+
+
+class EditorCache(object):
+    def __init__(
+        self,
+        recid,
+        record_version,
+        user_email=None,
+        task_name=None,
+    ):
+        self.recid = recid
+        self.record_version = record_version
+        self.user_email = user_email
+        self.task_name = task_name
+
+    @property
+    def redis(self):
+        redis_url = current_app.config.get("CACHE_REDIS_URL")
+        redis = StrictRedis.from_url(redis_url)
+        return redis
+
+    @property
+    def user_hash_name(self):
+        return f"editor-lock:{self.recid}@{self.record_version}"
+
+    @property
+    def task_hash_name(self):
+        return f"editor-task-lock:{self.recid}@{self.record_version}"
+
+    @property
+    def user_lock_message(self):
+        return f"{self.user_email} at {datetime.now().isoformat()}"
+
+    @property
+    def task_lock_message(self):
+        return f"{self.task_name} at {datetime.now().isoformat()}"
+
+    def add_user_lock(self):
+        self.redis.hset(
+            self.user_hash_name, key=self.user_email, value=self.user_lock_message
+        )
+        self.redis.expire(self.user_hash_name, time=REDIS_KEY_VALUE_PAIR_EXPIRE_TIME)
+
+    def add_task_lock(self):
+        self.redis.hset(
+            self.task_hash_name, key=self.task_name, value=self.task_lock_message
+        )
+        self.redis.expire(self.task_hash_name, time=REDIS_KEY_VALUE_PAIR_EXPIRE_TIME)
+
+    def remove_task_lock(self):
+        self.redis.hdel(self.task_hash_name, self.task_name)
+
+    def remove_user_lock(self):
+        self.redis.hdel(self.user_hash_name, self.user_email)
+
+    def prepare_editor_lock_api_payload(self):
+        payload = {}
+        existing_editor_user_locks = ", ".join(
+            [
+                value.decode()
+                for value in self.redis.hgetall(self.user_hash_name).values()
+            ]
+        )
+        existing_editor_task_locks = ", ".join(
+            [
+                value.decode()
+                for value in self.redis.hgetall(self.task_hash_name).values()
+            ]
+        )
+        if existing_editor_user_locks:
+            editor_user_lock_msg = f"The record might still be edited by {existing_editor_user_locks}".strip(
```

Comment:
could you use the full message as in the issue?

Reply:
This message doesn't fit many locks for same record.

Reply:
Good point. Maybe something like
```suggestion
            editor_user_lock_msg = f"Record opened by {existing_editor_user_locks}. Their modifications aren't saved yet, they (or yours) might get lost.".strip(
```
so they get the consequences.